library/core/src/array/iter.rs

```diff
@@ -132,6 +132,15 @@ impl<T, const N: usize> Iterator for IntoIter<T, N> {
     fn last(mut self) -> Option<Self::Item> {
         self.next_back()
     }
+
+    #[inline]
+    unsafe fn __iterator_get_unchecked(&mut self, idx: usize) -> Self::Item
+    where
+        Self: TrustedRandomAccess,
+    {
+        // Safety: Callers are only allowed to pass an index that is in bounds
```

Comment:
And the same index must not be passed twice, but that a) doesn't matter right now because `Copy` is required and b) without `Copy` the `Drop` impl would cause a potential double-free here as `self.alive.start` is not updated.

Might be worth mentioning that here in case someone in the future wants to lift the `Copy` requirement.

Reply:
> And the same index must not be passed twice,

That's actually allowed for iterators which are clonable.

> b) without Copy the Drop impl would cause a potential double-free here as self.alive.start is not updated.

Yes, that's the critical part.